### DISCUSSION TITLE
Unable to make it compatiable with latest version of pdfjs-dist

### DISCUSSION THREAD

**Discussion by Rahul1726260** [wojtekmaj/react-pdf#1581] [Q&A] [answered]
### Before you start - checklist

- [X] I followed instructions in documentation written for my React-PDF version
- [X] I have checked if this bug is not already reported
- [X] I have checked if an issue is not listed in [Known issues](https://github.com/wojtekmaj/react-pdf/wiki/Known-issues)
- [X] If I have a problem with PDF rendering, I checked if my PDF renders properly in [PDF.js demo](https://mozilla.github.io/pdf.js/web/viewer.html)

### Description

Failed to load pdf

: UnknownErrorException: The API version "3.6.172" does not match the Worker version "3.9.179".

### Steps to reproduce

load a pdf with similar version

### Expected behavior

Able to load pdf

### Actual behavior

Not able to load pdf

### Environment

- **Browser (if applicable)**:
- **React-PDF version**:
- **React version**:
- **Webpack version (if applicable)**:
   "pdfjs-dist": "^3.9.179",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-pdf": "^7.3.2",

**Comment by Rahul1726260** [comment]
Anyone can please help  with compatability issue of react-pdf and pdf-js in react project

**Comment by wojtekmaj** [comment, marked as answer]
If you have multiple pdfjs-dist versions in your app (I'd advise you not to), default ways of copying/including worker file won't work, because:

When having one copy of pdfjs-dist, it will look like this:

```
\- node_modules
   |- pdfjs-dist (3.6)
   \- react-pdf
```

When having multiple copies, it will look like this:

```
\- node_modules
   |- pdfjs-dist (in your case: 3.9)
   |  \- node_modules
   |     \- pdfjs-dist (in your case: 3.6)
   \- react-pdf
```

When importing `pdfjs-dist` directly from your application, you'll import version 3.9, not 3.6.

* If you copy the worker file using Webpack, Vite, or a custom script, you will need to change the directory you're copying from to node_modules/react-pdf/node_modules/pdfjs-dist (as opposed to node_modules/pdfjs-dist).
* You could also get worker file from external CDN (see README), which should work just fine.

**Reply by Rahul1726260** [reply]
Did the same and even I'm  having only one pdfjs-dist version installed  but still it shows a warning  : UnknownErrorException: The API version "3.6.172" does not match the Worker version "3.9.179".

**Reply by wojtekmaj** [reply]
Well clearly you still have pdfjs-dist installed in your repository. Use [yarn why](https://yarnpkg.com/cli/why) or similar command offered by whatever package manager you're using to figure out why you still have it.

**Reply by Rahul1726260** [reply]
I'm using npm and while installing pdfjs-dist  it install the latest package but react-pdf is useing 3.6.179

**Reply by wojtekmaj** [reply]
Don't install pdfjs-dist manually if you don't use it.

**Reply by Rahul1726260** [reply]
after unistalling pdfjs-dist and   going with solution provided in readme 

pdfjs.GlobalWorkerOptions.workerSrc = new URL(
  'pdfjs-dist/build/pdf.worker.min.js',
  import.meta.url,
).toString();

I'm not able to load the pdf  it always says failed to load pdf

**Reply by wojtekmaj** [reply]
There can be multiple reasons for PDF failing to load, most common are invalid URLs (check Network tab in dev tools for any 404s), and CORS issues (check Console tab in dev tools for any errors/warnings).

**Reply by Rahul1726260** [reply]
no cors nothing in network ,  as i install pdfjs-dist  wih the specofied verison eveything works fine

**Reply by attilaaronnagy** [reply]
@types/react-pdf is the one causing the problem (at least for me, if you update that package it resolves the issue)

**Reply by wojtekmaj** [reply]
You should not update it, you should REMOVE it. It's irrelevant now.

**Reply by attilaaronnagy** [reply]
yeah, I realised and removed it, node_modules reinstall and everything works. however if somebody just upgrades their package.json they we'll receive this problem upgrading from <7. probably some docs, that hey if you upgrade remove the @types package would be nice.
